chirp/drivers/th_uv88.py

```diff
@@ -1094,6 +1261,496 @@ def get_settings(self):
+                                         self.VALID_CHARS)
+            rset = RadioSetting(name, display, rx)
+            rset.set_apply_callback(_set_namelen, obj, atrb)
+            setting.append(rset)
+
+        def _tel_encode(char):
+            for i in range(len(OPTSIGTELDIGIT_LIST)):
+                if OPTSIGTELDIGIT_LIST[i] == char:
+                    return i
+            return 0
+
+        def _set_hexlen(setting, obj, atrb, maxlen):
+            vx = str(setting.value)
+            vx = vx.rstrip()
+            vxlen = len(vx)
+            vx = vx.ljust(maxlen, "0")
+            hval = []
+            LOG.debug(("HEXALEN %d:" % maxlen)+vx)
+            for i in range(maxlen // 2):
+                hval.append(_tel_encode(vx[2*i]) * 16 + _tel_encode(vx[2*i+1]))
+
+            if (maxlen % 2) == 1:
+                setattr(obj, atrb+"last", _tel_encode(vx[maxlen-1]))
+
+            setattr(obj, atrb, hval)
+            setattr(obj, atrb+"len", vxlen)
+            return
+
+        def _add_hexlen(setting, name, display, obj, atrb, maxlen,
+                        value, valuelen):
+            valname = ""
+            for i in range(valuelen):
+                if (i // 2) < len(value):
+                    if (i % 2) == 0:
+                        digit = int(value[i // 2]) // 16
+                    else:
+                        digit = int(value[i // 2]) % 16
+                else:
+                    digit = int(getattr(obj, atrb+"last"))
+
+                # Convert
+                valname += OPTSIGTELDIGIT_LIST[digit]
+
+            rx = RadioSettingValueString(0, maxlen, valname, False,
+                                         CHARSET_TEL)
+            rset = RadioSetting(name, display, rx)
+            rset.set_apply_callback(_set_hexlen, obj, atrb, maxlen)
+            setting.append(rset)
+
+        def _add_extdigit(setting, name, display, value, addoff):
+            digits = OPTSIGEXTDIGIT_LIST.copy()
+            if addoff:
+                digits.insert(0, "OFF")
+            digit = digits[value]
+            rx = RadioSettingValueList(digits, digit)
+            rset = RadioSetting(name, display, rx)
+            setting.append(rset)
+
+        def _add_ani(setting, name, display, value):
+            ani = OPTSIGENCTYPE_LIST[value]
+            rx = RadioSettingValueList(OPTSIGENCTYPE_LIST, ani)
+            rset = RadioSetting(name, display, rx)
+            setting.append(rset)
+
+        def _add_time(setting, name, display, start, step, count,
+                      format, value):
+            options = []
+            for i in range(count):
+                options.append(format % (start+i*step))
+            rx = RadioSettingValueList(options, options[value])
+            rset = RadioSetting(name, display, rx)
+            setting.append(rset)
+
+        def _set_int10(setting, obj, atrb):
+            vx = float(str(setting.value))
+            vx = int(vx * 10)
```

Comment:
I don't think you should need these `int` and `float` casts here. If you do, maybe there's some work that needs doing elsewhere.

Reply:
You are right. Removing the cast do the job.